pynxtools_mpes/reader.py

```diff
@@ -48,90 +49,75 @@
     "ky": "1/A",
 }
 
+def recursive_parse_metadata(
+    node: Union[h5py.Group, h5py.Dataset],
+) -> dict:
+    """Recurses through an hdf5 file, and parse it into a dictionary.
+
+    Args:
+        node (Union[h5py.Group, h5py.Dataset]): hdf5 group or dataset to parse into
+            dictionary.
 
-def res_to_xarray(res, bin_names, bin_axes, metadata=None):
-    """creates a BinnedArray (xarray subclass) out of the given np.array
-    Parameters:
-        res: np.array
-            nd array of binned data
-        bin_names (list): list of names of the binned axes
-        bin_axes (list): list of np.arrays with the values of the axes
     Returns:
-        ba: BinnedArray (xarray)
-            an xarray-like container with binned data, axis, and all
-            available metadata
+        dict: Dictionary of elements in the hdf5 path contained in node
     """
-    dims = bin_names
-    coords = {}
-    for name, vals in zip(bin_names, bin_axes):
-        coords[name] = vals
-
-    xres = xr.DataArray(res, dims=dims, coords=coords)
+    if isinstance(node, h5py.Group):
+        dictionary = {}
+        for key, value in node.items():
+            dictionary[key] = recursive_parse_metadata(value)
 
-    for name in bin_names:
+    else:
+        entry = node[...]
         try:
-            xres[name].attrs["unit"] = DEFAULT_UNITS[name]
-        except KeyError:
-            pass
-
-    xres.attrs["units"] = "counts"
-    xres.attrs["long_name"] = "photoelectron counts"
+            dictionary = entry.item()
+            if isinstance(dictionary, (bytes, bytearray)):
+                dictionary = dictionary.decode()
+        except ValueError:
+            dictionary = entry
 
-    if metadata is not None:
-        xres.attrs["metadata"] = metadata
+    return dictionary
 
-    return xres
 
+def h5_to_xarray(faddr: str, mode: str = "r") -> xr.DataArray:
+    """Read xarray data from formatted hdf5 file
 
-def h5_to_xarray(faddr, mode="r"):
-    """Rear xarray data from formatted hdf5 file
     Args:
         faddr (str): complete file name (including path)
-        mode (str): hdf5 read/write mode
+        mode (str, optional): hdf5 read/write mode. Defaults to "r".
+
+    Raises:
+        ValueError: Raised if data or axes are not found in the file.
+
     Returns:
-        xarray (xarray.DataArray): output xarra data
+        xr.DataArray: output xarra data
     """
     with h5py.File(faddr, mode) as h5_file:
         # Reading data array
         try:
-            data = h5_file["binned"]["BinnedData"]
-        except KeyError:
-            print("Wrong Data Format, data not found")
-            raise
+            data = np.asarray(h5_file["binned"]["BinnedData"])
+        except KeyError as exc:
+            raise ValueError(
+                f"Wrong Data Format, the BinnedData were not found. The error was{exc}.",
```

Comment:
```suggestion
                f"Wrong Data Format, the BinnedData were not found.",
```

I'm not sure but isn't the exc message of the original raising message included in the error trace when you do `... from exc`?